azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NetworkWatcher.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.network;
+
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.network.implementation.NetworkManager;
+import com.microsoft.azure.management.network.implementation.NetworkWatcherInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.GroupableResource;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.Resource;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
+
+/**
+ * Entry point for Network Watcher API in Azure.
+ */
+@Fluent
+@Beta
+public interface NetworkWatcher extends
+        GroupableResource<NetworkManager, NetworkWatcherInner>,
+        Refreshable<NetworkWatcher>,
+        Updatable<NetworkWatcher.Update> {
+
+    /**
+     * @return entry point to manage packet captures associated with network watcher
+     */
+    PacketCaptures packetCaptures();
+
+    /**
+     * @param targetResourceGroup the name of the target resource group to perform getTopology on
+     * @return current network topology by resource group
+     */
+    Topology getTopology(String targetResourceGroup);
```

Comment:
To make it consistent with the portal, it might make sense to leave it as Topology. This is also a method on NetworkWatcher object, so it seems excessive to call it getNetworkTopology.